command/agent/auth/azure/azure.go

```diff
@@ -176,3 +175,33 @@ func getMetadataInfo(ctx context.Context, endpoint, resource string) ([]byte, er
 
 	return body, nil
 }
+
+func verifyIdentity(ctx context.Context) (retPath string, header http.Header, retData map[string]interface{}, retErr error) {
+
+	// after given IDMS given access token we want to only auth with a particular given client and object id
+	// example curl https://management.azure.com/subscriptions/<SUBSCRIPTION ID>/resourceGroups/<RESOURCE GROUP>?api-version=2016-09-01 -H "Authorization: Bearer <ACCESS TOKEN>"
+
+	// Fetch instance data
+	var instance struct {
+		Compute struct {
+			Name              string
+			ResourceGroupName string
+			SubscriptionID    string
+			VMScaleSetName    string
+		}
+	}
+
+	body, err := getMetadataInfo(ctx, instanceEndpoint, "")
+	if err != nil {
+		retErr = err
+		return
+	}
+
+	err = jsonutil.DecodeJSON(body, &instance)
```

Comment:
It looks like we decode the JSON body into instance, but the object is scoped to this function, should that be returned instead? I also see that `retData` remains empty and not assigned, should this be set somewhere?